sql/core/src/test/scala/org/apache/spark/sql/sources/DDLTestSuite.scala

```diff
@@ -113,4 +124,20 @@ class DDLTestSuite extends DataSourceTest with SharedSQLContext {
     assert(attributes.map(_.name) === Seq("col_name", "data_type", "comment"))
     assert(attributes.map(_.dataType).toSet === Set(StringType))
   }
+
+  test("SPARK-7012 Create table statement should support NOT NULL modifier for columns") {
+    sql(
```

Comment:
you can use `withTempPath` for this test. [example](https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala#L1732-L1765)